src/store/components/AdminItemPage/index.tsx

```diff
@@ -21,30 +22,43 @@ interface AdminItemPageProps {
   collections: PublicMerchCollection[];
 }
 
-interface AdminItemPageForm {
-  collection?: string;
-  itemName?: string;
-  description?: string;
-  existingPicture?: string;
-  newPicture?: Blob;
-  hidden?: boolean;
-  monthlyLimit?: number;
-  lifetimeLimit?: number;
-  hasVariantsEnabled?: boolean;
-
-  categoryName?: string;
-  options?: {
-    uuid?: Uuid;
-    value: string;
-    price: string;
-    quantity: string;
-    discountPercentage: string;
-  }[];
-
-  quantity?: string;
-  price?: string;
-  discountPercentage?: string;
-}
+const AdminItemPageFormSchema = Yup.object().shape({
+  collection: Yup.string().required('Required'),
+  itemName: Yup.string().min(2, 'Too Short').max(50, 'Too Long').required('Required'),
+  description: Yup.string().min(2, 'Too Short').max(1000, 'Too Long').required('Required'),
+  // hidden: Yup.boolean().required('Required'),
+  monthlyLimit: Yup.number().min(1, 'Too Low').max(10, 'Too High').required('Required'),
+  lifetimeLimit: Yup.number().min(1, 'Too Low').max(10, 'Too High').required('Required'),
+  // hasVariantsEnabled: Yup.boolean().required('Required'),
+  categoryName: Yup.string().when('hasVariantsEnabled', {
+    is: (hasVariantsEnabled) => hasVariantsEnabled,
+    then: Yup.string().min(2, 'Too Short').max(50, 'Too Long').required('Required'),
+  }),
+  options: Yup.array().when('hasVariantsEnabled', {
+    is: (hasVariantsEnabled) => hasVariantsEnabled,
+    then: Yup.array().of(
+      Yup.object().shape({
+        uuid: Yup.string(),
+        value: Yup.string().min(2, 'Too Short').max(50, 'Too Long').required('Required'),
+        price: Yup.number().min(1, 'Too Low').max(1000000, 'Too High').required('Required'),
+        quantity: Yup.number().min(1, 'Too Low').max(1000000, 'Too High').required('Required'),
```

Comment:
won't this make it impossible to edit a sold out item? i feel like we should allow quantities of 0 (to signify sold out?)

idk im not sure how it works on the backend after an item is purchased (does quantity get modified?)

Reply:
I'm going to address this issue in a separate PR. The way backend handles quantity updates is via a separate route that only allows positively incrementing quantities, so I'm going to ask about changing that, and then will address what the valid field values are.